content/en/blog/2021/wasm-api-alpha/index.md

```diff
@@ -0,0 +1,119 @@
+    someSetting: true
+    someOtherSetting: false
+    youNameIt:
+    - first
+    - second
+  url: docker.io/your-org/your-filter:1.0.0
+{{< /text >}}
+
+There are a lot of similarities and a few differences between WasmPlugin and `EnvoyFilter`, so let’s go through the fields one by one. The above example deploys a Wasm module to all workloads (including gateway pods) that match it’s `selector` - this very much works the same as in an `EnvoyFilter`.
+
+The next field below that is the `phase`. This determines where in the proxy’s filter chain the Wasm module will be injected. We have defined four distinct phases for injection:
+
+* `AUTHN`: prior to any Istio authentication and authorization filters.
+* `AUTHZ`: after the Istio authentication filters and before any first class authorization filters, i.e. before AuthorizationPolicies have been applied.
+* `STATS`: after all authorization filters and prior to the Istio stats filter.
+* `UNSPECIFIED_PHASE`: let the control plane decide where to insert. This will generally be at the end of the filter chain, right before the Router. This is the default value for this `phase` field.
+
+`pluginConfig` is used for configuring your Wasm plugin. Whatever you put into this field will be encoded in JSON and passed on to your filter, where you can access it in the configuration callback of the Proxy-Wasm SDKs. For example, you can retrieve the config on `onConfigure` in [C++ SDK](https://github.com/proxy-wasm/proxy-wasm-cpp-sdk/blob/fd0be8405db25de0264bdb78fae3a82668c03782/proxy_wasm_api.h#L329-L331), `on_configure` in [Rust SDK](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/blob/v0.1.4/src/dispatcher.rs#L255) or `OnPluginStart` call back in [Go SDK](https://github.com/tetratelabs/proxy-wasm-go-sdk/blob/v0.15.0/proxywasm/types/context.go#L74).
+
+The `url` field specifies where to pull the Wasm module. You’ll notice that the `url` in this case is a docker URI - this is because apart from loading Wasm modules via HTTP, HTTPS and the local file system (using file://), we are introducing the OCI image format as the preferred mechanism for distributing Wasm modules.
+
+One last thing to note is currently the Wasm Plugin API only applies to inbound HTTP filter chains.
+Adding support for network filters and outbound traffics is the future work.
+
+## Wasm image specification
+
+We believe that containers are the ideal way to store, publish and manage proxy extensions, so we worked with Solo.io to extend their existing Proxy-Wasm container format with a variant that aims to be compatible with all registries and CLI toolchain. Depending on your processes, you can now build your proxy extension containers using your existing container CLI tooling such as Docker CLI or [buildah](https://buildah.io/).
+
+For detail, please refer to [the link here](https://github.com/istio-ecosystem/wasm-extensions/blob/master/doc/how-to-build-oci-images.md), and learn how to build OCI images that are consumable by Istio agent.
```

Comment:
Maybe make this something like: 

To learn how to build OCI images, please refer to this [link]().